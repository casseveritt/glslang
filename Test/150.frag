#version 150 core

in vec4 gl_FragCoord;
layout(origin_upper_left, pixel_center_integer) in vec4 gl_FragCoord;  // ERROR
layout(pixel_center_integer) in vec4 gl_FragCoord;  // ERROR
layout(origin_upper_left) in vec4 foo;  // ERROR
layout(origin_upper_left, pixel_center_integer) in vec4 gl_FragCoord;

void main()
{
    vec4 c = gl_FragCoord;
}

layout(origin_upper_left, pixel_center_integer) in vec4 gl_FragCoord;  // ERROR, declared after use

in struct S { float f; } s;

float patch = 3.1;