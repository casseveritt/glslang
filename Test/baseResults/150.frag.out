150.frag
Warning, version 150 is not yet complete; some version-specific features are present, but many are missing.
ERROR: 0:4: 'redeclaration' : cannot redeclare with different qualification: gl_FragCoord
ERROR: 0:5: 'redeclaration' : cannot redeclare with different qualification: gl_FragCoord
ERROR: 0:6: 'layout qualifier' : can only apply origin_upper_left and pixel_center_origin to gl_FragCoord 
ERROR: 0:14: 'gl_FragCoord' : cannot redeclare after use 
ERROR: 4 compilation errors.  No code generated.


gl_FragCoord pixel center is integer
gl_FragCoord origin is upper left
ERROR: node is still EOpNull!
0:9  Function Definition: main( (void)
0:9    Function Parameters: 
0:11    Sequence
0:11      Sequence
0:11        move second child to first child (4-component vector of float)
0:11          'c' (4-component vector of float)
0:11          'gl_FragCoord' (gl_FragCoord 4-component vector of float)
0:18  Sequence
0:18    move second child to first child (float)
0:18      'patch' (float)
0:18      Constant:
0:18        3.100000
0:?   Linker Objects
0:?     'gl_FragCoord' (gl_FragCoord 4-component vector of float)
0:?     'gl_FragCoord' (gl_FragCoord 4-component vector of float)
0:?     'foo' (smooth in 4-component vector of float)
0:?     's' (smooth in structure{float f})
0:?     'patch' (float)


Linked fragment stage:


gl_FragCoord pixel center is integer
gl_FragCoord origin is upper left

